### PR TITLE
Task/14347 Add SRS error codes

### PR DESCRIPTION
**Pull request by 30mar**
## Description
add srs errors codes

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-14347

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Services/PPAccessControl/Model/PPACError.swift [hunk 1]
```diff
@@ -20,6 +22,11 @@ enum PPACError: Error {
 			return "timeIncorrect"
 		case .timeUnverified:
 			return "timeUnverified"
+		case .minimumTimeSinceOnboarding:
+			return "minimumTimeSinceOnboarding"
```
Comment (flxschmidt) [inline src/xcode/ENA/ENA/Source/Services/PPAccessControl/Model/PPACError.swift]: `"TIME_SINCE_ONBOARDING_UNVERIFIED"` as raw value?

from https://github.com/corona-warn-app/cwa-app-tech-spec/blob/proposal/self-report/docs/spec/data-donation-client-srs.md#submitting-a-self-report

**Review by flxschmidt — CHANGES_REQUESTED** [review]
Only one thing :)

File: src/xcode/ENA/ENA/Source/Services/PPAccessControl/Model/PPACError.swift (hunk 1 above)
Comment (naveeddotio) [reply src/xcode/ENA/ENA/Source/Services/PPAccessControl/Model/PPACError.swift]: What about the above values like timeUnverified? can we follow the same pattern there?

File: src/xcode/ENA/ENA/Source/Services/PPAccessControl/Model/PPACError.swift [hunk 2]
```diff
@@ -9,21 +9,21 @@ enum PPACError: Error {
...
+	case submissionTooEarly
 	
 	var description: String {
 		switch self {
 		case .generationFailed:
-			return "deviceCheck Token generation failed"
+			return "DEVICE_TOKEN_GENERATION_FAILED"
 		case .deviceNotSupported:
-			return "deviceNotSupported"
+			return "DEVICE_TOKEN_NOT_SUPPORTED"
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Services/PPAccessControl/Model/PPACError.swift:20]: DEVICE or DEVICE_TOKEN ?

File: src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/Survey/SurveyURLProvider.swift [hunk 3]
```diff
@@ -51,6 +57,10 @@ enum SurveyError: Error {
 			return String(format: AppStrings.SurveyConsent.errorChangeDeviceTime, errorCode)
 		case .alreadyParticipated(let errorCode):
 			return String(format: AppStrings.SurveyConsent.errorAlreadyParticipated, errorCode)
+		case .submissionTooEarly, .minimumTimeSinceOnboarding:
+			//these cases are un related to the survay
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/Survey/SurveyURLProvider.swift]: // these cases are unrelated to the survey

File: src/xcode/ENA/ENA/Source/Services/PPAccessControl/Model/PPACError.swift (hunk 2 above)
Comment (30mar) [reply src/xcode/ENA/ENA/Source/Services/PPAccessControl/Model/PPACError.swift:20]: it should be device token

File: src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/Survey/SurveyURLProvider.swift (hunk 3 above)
Comment (30mar) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/Survey/SurveyURLProvider.swift]:
```suggestion
			/ /these cases are unrelated to the survey
```

File: src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/Survey/SurveyURLProvider.swift [hunk 4]
```diff
@@ -51,6 +57,10 @@ enum SurveyError: Error {
 			return String(format: AppStrings.SurveyConsent.errorChangeDeviceTime, errorCode)
 		case .alreadyParticipated(let errorCode):
 			return String(format: AppStrings.SurveyConsent.errorAlreadyParticipated, errorCode)
+		case .submissionTooEarly, .minimumTimeSinceOnboarding:
+			/ /these cases are unrelated to the survey
+			return ""
+			break
```
Comment (30mar) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/Survey/SurveyURLProvider.swift]:
```suggestion

```

File: src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/Survey/SurveyURLProvider.swift [hunk 5]
```diff
@@ -51,6 +57,10 @@ enum SurveyError: Error {
 			return String(format: AppStrings.SurveyConsent.errorChangeDeviceTime, errorCode)
 		case .alreadyParticipated(let errorCode):
 			return String(format: AppStrings.SurveyConsent.errorAlreadyParticipated, errorCode)
+		case .submissionTooEarly, .minimumTimeSinceOnboarding:
+			/ /these cases are unrelated to the survey
```
Comment (30mar) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/Survey/SurveyURLProvider.swift]:
```suggestion
			// these cases are unrelated to the survey
```

**Review by naveeddotio — APPROVED** [review]

**Review by flxschmidt — APPROVED** [review]
LGTM!
